### ISSUE TITLE
Use ARM64 for docker deployment job

### ISSUE THREAD

**Issue by pawlh** [softwareconstruction240/autograder#384]
GitHub has recently made ARM64 runners available, removing the need to use `qemu` when building for the production deployment

https://github.blog/2024-06-03-arm64-on-github-actions-powering-faster-more-efficient-build-systems/

The only reason I can see why this would not be feasible is if `softwareconstruction240`'s GitHub plan is incompatible

**Comment by 19mdavenport** [comment]
This availability is still in public beta. We will need to wait until this feature is fully released, but good to keep in mind: https://github.com/github/roadmap/issues/970

**Comment by 19mdavenport** [comment]
I probably should have checked this a while ago, but looks like this might be ready: https://github.blog/changelog/2025-01-16-linux-arm64-hosted-runners-now-available-for-free-in-public-repositories-public-preview/

**Comment by TheDavSmasher** [comment]
@19mdavenport So, what does the change entail, exactly? Or, rather, how to do it I mean

**Comment by 19mdavenport** [comment]
@TheDavSmasher The linked article should be able to help somewhat. Otherwise, I'm not completely sure atm. @pawlh if you see this, could you give a slightly more detailed description of why you wanted this for the autograder?

**Comment by pawlh** [comment]
Yes- This is relevant because the EC2 instance that actually runs the autograder uses ARM64 as its architecture. When I first created the docker build steps, the only runners available were AMD, meaning I had to use an emulator in the workflow to allow it to build for a different architecture

If you check out [this documentation](https://docs.github.com/en/actions/using-github-hosted-runners/using-github-hosted-runners/about-github-hosted-runners#standard-github-hosted-runners-for-public-repositories), you'll see that the only change that would need to be down is to change [this line](https://github.com/softwareconstruction240/autograder/blob/31b26893f30a68d32c22602187125cef926b0131/.github/workflows/docker-publish.yml#L14) to be `ubuntu-24.04-arm` instead of `ubuntu-latest`. Also, we'd simply delete the [qemu lines](https://github.com/softwareconstruction240/autograder/blob/31b26893f30a68d32c22602187125cef926b0131/.github/workflows/docker-publish.yml#L29)

I took a peek at the[ available runners](https://github.com/softwareconstruction240/autograder/actions/runners) for this project, but didn't see the ARM runners listed. That means either they haven't been rolled out to this repo yet or an admin for this repo will need to go into the action settings and add an ARM runner
